### PR TITLE
Basic implementations of nanoTime and currentTimeMillis

### PR DESCRIPTION
**Pull request by andreaTP**
This is an initial basic implementation of System.currentTimeMillis and System.nanoTime
results are also incredibly similar XD

note: I put constants into the separate object as suggested by @sjrd

### CODE REVIEW

**Comment by densh** [comment]
Please format your code with scalafmt.

**Comment by andreaTP** [comment]
done and rebased

**Comment by densh** [comment]
Needs another rebase on top of latest master due to CI changes.

**Comment by andreaTP** [comment]
should be ok

File: nativelib/src/main/scala/scala/scalanative/native/time.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  @struct
+  class timespec(
+      val tv_sec: time_t = 0L, /* seconds */
+      val tv_nsec: CLong = 0L /* nanoseconds */
+  )
+
+  def clock_getres(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt  = extern
+  def clock_gettime(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt = extern
+  def clock_settime(clk_id: Clock.clockid_t, res: Ptr[timespec]): CInt = extern
```
Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: It looks like those are [not C11 libc functions](http://www.cplusplus.com/reference/ctime/), but posix time functions. `@extern object time` should only contain the bindings to things defined in `time.h`. We also can't unconditionally depend on posix in our core libraries as that would complicate Windows support.

Comment (andreaTP) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: ok, given that there is not a way to achieve the desired resolution in a complete OS/independent way....
the question is: do you prefer to not support this kind of functions all around or you want to support OS/dependent compilation?

Unfortunately this apply to many other areas, files etc. etc. Posix systems are often not compatible with Windows.

Comment (andreaTP) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: i.e.
the equivalent of ```clock_gettime(CLOck_MONOTONIC, ptr)``` on widows is ```QueryPerformanceCounter()```

Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: Once https://github.com/scala-native/scala-native/issues/58 lands we'll be able to implement subsets of APIs in platform-dependent manner (e.g. cross project that has two separate implementations for windows and unix-like systems). For now we should avoid using any platform-specific APIs as that would make it hard to get rid of those dependencies later on.

Comment (andreaTP) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: Couldn't we tag with a specific comment the related parts of code?
In this way when we will have #58 merged we can move the code to the proper subproject.

Another option is to start to have a couple of subprojects like ```posix-nativelib``` and ```windows-nativelib``` that can be switched in user build imports but that respect the very same classes/methods signatures.

Probably people implementing things on unix and win environment are potentially different (i.e. I have no win dev machine to test on).

Comment (andreaTP) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: As a possible temporary workaround we could also check:
https://github.com/ThoughtWorksInc/enableIf.scala

Comment (densh) [inline nativelib/src/main/scala/scala/scalanative/native/time.scala]: I'll prioritise #58. Lets just have posix implementation in master for now. We probably don't want to expose those as public part of SN yet, so lets put bindings to that in `scala.scalanative.runtime` under `@extern object posix`. Just make sure that you use only cross-platform posix APIs and don't use linux-only things.

**Comment by andreaTP** [comment]
I haven't catch the point, you wish to accept a posix implementation but you prefer to avoid binding it to SN itself?
This means that I should not bind ```java.lang.System.``` methods...

btw can you share the steps to keep cross projects works, somebody could help maybe...

Cheers!

**Comment by densh** [comment]
What I meant in https://github.com/scala-native/scala-native/pull/227#discussion_r70049954 is:

1. Assume posix-compatible OS like Linux, MacOS and FreeBSD (that's what people have been using SN on so far.)
2. Put bindings for posix-specific things into `scala.scalanative.runtime` package as it's not a public API we support yet but just an implementation detail. 
3. Use those bindings to implement parts of Java library like `nanoTime` and  `currentTimeMillis`.

In the future we'll refactor those in platform-independent manner.

**Comment by andreaTP** [comment]
unfortunately after a little search seems like clock_gettime, is not implemented on osX even if it is posix ...
I wait for cross building spread out to see hoe this can work :-S

**Comment by densh** [comment]
Lets pause this one for now are re-open once cross-projects are done.

**Comment by andreaTP** [comment]
sounds good

**Pull request closed without merging**